lib/Sema/CSSimplify.cpp

```diff
@@ -6453,6 +6453,13 @@ bool ConstraintSystem::repairFailures(
                             getConstraintLocator(anchor, path));
     }
 
+    if (!path.empty()) {
+      if (path[path.size() - 1].is<LocatorPathElt::ApplyArgToParam>())
```

Comment:
Just a nit: `path` has a `back()` for last elt. 
```suggestion
      if (path.back().is<LocatorPathElt::ApplyArgToParam>())
```

Reply:
:)